### PR TITLE
PlatformIO: Fix ESP-IDF generation. (fix #734)

### LINKED ISSUE #734: Platform IO with platform espressif32 includes generated sources twice
**Steps to reproduce the issue**
* Create platformio with below platformio.ini
* Use nanopb version 0.4.6.2

```ini
[env:firebeetle32]
platform = espressif32
board = firebeetle32
framework = espidf
lib_deps = nanopb/Nanopb
nanopb_protos = +<protos/test.proto>
```

**What happens?**
Platform IO includes generated source files twice when including nanopb in a project targetting espressif32.

Output of Build
```log
Warning! Ignore unknown configuration option `nanopb_protos` in section [env:firebeetle32]
Processing firebeetle32 (platform: espressif32; board: firebeetle32; framework: espidf)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/firebeetle32.html
PLATFORM: Espressif 32 (3.4.0) > FireBeetle-ESP32
HARDWARE: ESP32 240MHz, 320KB RAM, 4MB Flash
DEBUG: Current (esp-prog) External (esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-espidf 3.40301.0 (4.3.1)
 - tool-cmake 3.16.4
 - tool-esptoolpy 1.30100.210531 (3.1.0)
 - tool-idf 1.0.1
 - tool-mconf 1.4060000.20190628 (406.0.0) 
 - tool-ninja 1.9.0
 - toolchain-esp32ulp 1.22851.191205 (2.28.51)
 - toolchain-riscv32-esp 8.4.0+2021r1
 - toolchain-xtensa-esp32 8.4.0+2021r1
 - toolchain-xtensa-esp32s2 8.4.0+2021r1
Reading CMake configuration...
Installing Protocol Buffers dependencies
Requirement already satisfied: protobuf>=3.19.1 in c:\users\user\.platformio\penv\lib\site-packages (3.19.1)
[nanopb] Skipping 'protos\test.proto' (no options)
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ soft
Installing Protocol Buffers dependencies
Requirement already satisfied: protobuf>=3.19.1 in c:\users\user\.platformio\penv\lib\site-packages (3.19.1)
[nanopb] Skipping 'protos\test.proto' (no options)

*** Two environments with different actions were specified for the same target: C:\Users\user\temp\nanopb_sample\.pio\build\firebeetle32\nanopb\generated-build\test.pb.o
(action 1: xtensa-esp32-elf-gcc -o test.pb.o -c -DPLATFORMIO=50204 -DARDUINO_ESP32_DEV test.pb.c)
(action 2: xtensa-esp32-elf-gcc -o test.pb.o -c -Og -Wall -Werror=all -Wextra -Wno-error=deprecated-declarations -Wno-error=unused-but-set-variable -Wno-error=unused-function -Wno-error=unused-variable -Wno-frame-address -Wno-old-style-declaration -Wno-sign-compare -Wno-unused-parameter -fdata-sections -ffunction-sections -fstrict-volatile-bitfields -ggdb -mlongcalls -std=gnu99 -D_GNU_SOURCE -DIDF_VER=\"4.3.1\" -DESP_PLATFORM -DPLATFORMIO=50204 -DARDUINO_ESP32_DEV test.pb.c)
File "C:\Users\user\.platformio\penv\lib\site-packages\platformio\builder\tools\platformio.py", line 359, in <listcomp>
====================================================================================== [FAILED] Took 5.01 seconds ======================================================================================
The terminal process "C:\Users\user\.platformio\penv\Scripts\platformio.exe 'run'" terminated with exit code: 1.

Terminal will be reused by tasks, press any key to close it.
```

**What should happen?**
The generated sources should only be included once.

### PR DESCRIPTION
**Pull request by hacker-cb**

### CODE REVIEW

**Comment by hacker-cb** [comment]
@PetteriAimonen please wait for the CI finish before merge

File: generator/platformio_generator.py
Comment (mathisloge) [inline generator/platformio_generator.py:151]: `env['PIOENV']` could contain `-` which are not allowed with scons. So something like `env['PIOENV'].replace("-", "_")` should be there. Otherwise you get a `Illegal construction variable` error. 
See: https://github.com/SCons/scons/blob/master/test/bad-variables.py

**Review by mathisloge — CHANGES_REQUESTED** [review]

**Review by mathisloge — APPROVED** [review]
tested with my setup.

**Comment by PetteriAimonen** [comment]
Thanks!

**Comment by hacker-cb** [comment]
@PetteriAimonen could you please publish updated version to the PlatformIO registry?
Thanks in advance.

**Pull request merged**
